src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1098,6 +1186,146 @@ export default class ComparisonStore {
         },
     });
 
+    readonly genericAssayEnrichmentAnalysisGroupsGroupByGenericAssayType = remoteData(
+        {
+            await: () => [
+                this
+                    .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType,
+                this.enrichmentAnalysisGroups,
+            ],
+            invoke: () => {
+                return Promise.resolve(
+                    _.mapValues(
+                        this
+                            .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType
+                            .result!,
+                        selectedGenericAssayEnrichmentProfileMap => {
+                            let studyIds = Object.keys(
+                                selectedGenericAssayEnrichmentProfileMap
+                            );
+                            // assumes single study for now
+                            if (studyIds.length === 1) {
+                                return this.enrichmentAnalysisGroups.result!.reduce(
+                                    (
+                                        acc: EnrichmentAnalysisComparisonGroup[],
+                                        group
+                                    ) => {
+                                        // filter samples having mutation profile
+                                        const filteredSamples = group.samples.filter(
+                                            sample =>
+                                                selectedGenericAssayEnrichmentProfileMap[
+                                                    sample.studyId
+                                                ] !== undefined
+                                        );
+                                        if (filteredSamples.length > 0) {
+                                            acc.push({
+                                                ...group,
+                                                count: filteredSamples.length,
+                                                samples: filteredSamples,
+                                                description: `samples in ${group.name}`,
+                                            });
+                                        }
+                                        return acc;
+                                    },
+                                    []
+                                );
+                            } else {
+                                return [];
+                            }
+                        }
+                    )
+                );
+            },
+        }
+    );
+
+    readonly genericAssayEnrichmentDataRequestGroupsGroupByGenericAssayTypes = remoteData(
```

Comment:
`gaDataGroupedByAssayType`? I am not sure what this is

Reply:
can you please explain what this is?

Reply:
This is another map (keyed by `genericAssayType`), which prepares `group` parameter to be sent to `fetchGenericAssayEnrichmentsUsingPOST` API.

Reply:
thanks. how about `gaDataQueriesByAssayType`?

Reply:
Sounds good, I will rename them!